From 423d57fc0a3f48528da3cdc85088077b8cc5e34a Mon Sep 17 00:00:00 2001
From: Volodymyr Bendiuga <volodymyr.bendiuga@westermo.se>
Date: Fri, 1 Feb 2019 10:45:25 +0100
Subject: [PATCH 35/38] route/link: allocate memory for egress QoS
Organization: Westermo Network Technologies AB

Failing to do so makes it impossible to set egress QoS mapping
for newly created Vlan object.

Signed-off-by: Volodymyr Bendiuga <volodymyr.bendiuga@westermo.se>
Signed-off-by: Joachim Nilsson <troglobit@gmail.com>
---
 lib/route/link/vlan.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/route/link/vlan.c b/lib/route/link/vlan.c
index 7c5aa06..e0072f0 100644
--- a/lib/route/link/vlan.c
+++ b/lib/route/link/vlan.c
@@ -76,6 +76,10 @@ static int vlan_alloc(struct rtnl_link *link)
 		if ((vi = calloc(1, sizeof(*vi))) == NULL)
 			return -NLE_NOMEM;
 
+		vi->vi_negress = 0;
+		vi->vi_egress_size = 32;
+		vi->vi_egress_qos = calloc(vi->vi_egress_size, sizeof(*vi->vi_egress_qos));
+
 		link->l_info = vi;
 	}
 
-- 
2.25.1

